### PR TITLE
Added article endpoint

### LINKED ISSUE #8: Add endpoint to get a single article by id
Endpoint could be: `GET /articles/article/:id`

### PR DESCRIPTION
**Pull request by mahmoudal99**
* Added endpoint to get a single article by id
closes #8

### CODE REVIEW

**Comment by salahusama** [comment]
You should add: `Closes #8` in the description so the issue gets closed automatically once this is merged, @mahmoudal99

File: routes/article.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const express = require('express')
+const router = express.Router()
+
+router.get('articles/article/:id', (req, res) => {
+    const db = req.app.get('db')
+    db.query('SELECT * FROM article WHERE id = ' + req.params.id, (error, results) => {
```
Comment (salahusama) [inline routes/article.js]: This is actually cool. Concatenating the string like this allows people to run malicious SQL code on our DB. Check [SQL Injection](https://www.w3schools.com/sql/sql_injection.asp) if you're interested.

If I call: `/articles/article/150 OR 1=1`, I will get all the articles. Of course, I can run any SQL statement. Something like:
`/articles/article/150; DROP TABLE articles` will delete the whole table.

The [documentation](https://github.com/mysqljs/mysql#preparing-queries) shows you how to add variables safely.
```suggestion
    const sql = mysql.format('SELECT * FROM article WHERE id = ??', [req.params.id])
    db.query(sql, (error, results) => {
```

Ask if its not clear.

File: routes/article.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const express = require('express')
+const router = express.Router()
+
+router.get('articles/article/:id', (req, res) => {
```
Comment (salahusama) [inline routes/article.js]: You need a `/` at the start of the path for it to work.
```suggestion
router.get('/articles/article/:id', (req, res) => {
```

File: routes/article.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    const db = req.app.get('db')
+    db.query('SELECT * FROM article WHERE id = ' + req.params.id, (error, results) => {
+
```
Comment (salahusama) [inline routes/article.js]: Nitpick: I dont think we need this line.
```suggestion
```

**Comment by salahusama** [comment]
Also, @mahmoudal99, I don't this branch is up to date with master. You may need to run `git merge master` from your branch.

I'm not sure if you need this, but you may need to checkout master, pull to get remote changes, then checkout your branch, and merge master into it:
```
git checkout master
git pull
git checkout article_endpoint
git merge master
```

File: routes/article.js (hunk 3 above)
Comment (mahmoudal99) [reply routes/article.js]: Are you referring to the error throw?

Comment (salahusama) [reply routes/article.js]: No, the empty line.

**Comment by salahusama** [comment]
I thought this was merged, @mahmoudal99 ?

**Pull request merged**